crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -2110,6 +2110,8 @@ impl<'a> Checker<'a> {
             .flatten()
             .collect();
 
+        self.semantic.flags |= SemanticModelFlags::DUNDER_ALL_DEFINITION;
```

Comment:
I think we should set and unset this around the `self.semantic.add_global_reference` calls, just to be safe.